core/src/nodes/tasks/task.rs

```diff
@@ -268,10 +273,13 @@ where
                                 this.taken_over.push(take_over),
                             Poll::Ready(None) =>
                                 // Node closed by the external API; start closing.
-                                if let Some(n) = node {
-                                    this.state = State::Closing(n.close());
+                                if node.is_some() {
```

Comment:
Wouldn't this also work, given that `extract_node` returns `None` when there is no node?

```rust
                                if let Some(node) = this.state.extract_node() {
                                    this.state = State::Closing(node.close());
                                    continue 'poll
                                } else {
                                    this.state = State::Undefined;
                                    return Poll::Ready(()) // end task
                                }
```